### PR TITLE
Make SECRETS_RSTUF* a sub-settings of RSTUF_AUTH

### LINKED ISSUE #324: Bug: When using RSTUF_AUTH=false: SECRETS_RSTUF_TOKEN and SECRETS_RSTUF_ADMIN_PASSWORD still required
### What happened?

When using `RSTUF_AUTH=false`, the container still expects `SECRETS_RSTUF_TOKEN` and `SECRETS_RSTUF_ADMIN_PASSWORD` still required.

The `SECRETS_RSTUF_TOKEN` and `SECRETS_RSTUF_ADMIN_PASSWORD` should be sub-settings of `RSTUF_AUTH` when it is a `true`.

### What steps did you take?

Run RSTUF API using `RSTUF_AUTH=false` without setting `SECRETS_RSTUF_TOKEN` and `SECRETS_RSTUF_ADMIN_PASSWORD`

### What behavior did you expect?

_No response_

### Relevant log output

```shell
rstuf-api_1       | Traceback (most recent call last):
rstuf-api_1       |   File "/usr/local/bin/uvicorn", line 8, in <module>
rstuf-api_1       |     sys.exit(main())
rstuf-api_1       |   File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
rstuf-api_1       |     return self.main(*args, **kwargs)
rstuf-api_1       |   File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1055, in main
rstuf-api_1       |     rv = self.invoke(ctx)
rstuf-api_1       |   File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
rstuf-api_1       |     return ctx.invoke(self.callback, **ctx.params)
rstuf-api_1       |   File "/usr/local/lib/python3.10/site-packages/click/core.py", line 760, in invoke
rstuf-api_1       |     return __callback(*args, **kwargs)
rstuf-api_1       |   File "/usr/local/lib/python3.10/site-packages/uvicorn/main.py", line 403, in main
rstuf-api_1       |     run(
rstuf-api_1       |   File "/usr/local/lib/python3.10/site-packages/uvicorn/main.py", line 568, in run
rstuf-api_1       |     server.run()
rstuf-api_1       |   File "/usr/local/lib/python3.10/site-packages/uvicorn/server.py", line 59, in run
rstuf-api_1       |     return asyncio.run(self.serve(sockets=sockets))
rstuf-api_1       |   File "/usr/local/lib/python3.10/asyncio/runners.py", line 44, in run
rstuf-api_1       |     return loop.run_until_complete(main)
rstuf-api_1       |   File "/usr/local/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
rstuf-api_1       |     return future.result()
rstuf-api_1       |   File "/usr/local/lib/python3.10/site-packages/uvicorn/server.py", line 66, in serve
rstuf-api_1       |     config.load()
rstuf-api_1       |   File "/usr/local/lib/python3.10/site-packages/uvicorn/config.py", line 471, in load
rstuf-api_1       |     self.loaded_app = import_from_string(self.app)
rstuf-api_1       |   File "/usr/local/lib/python3.10/site-packages/uvicorn/importer.py", line 21, in import_from_string
rstuf-api_1       |     module = importlib.import_module(module_str)
rstuf-api_1       |   File "/usr/local/lib/python3.10/importlib/__init__.py", line 126, in import_module
rstuf-api_1       |     return _bootstrap._gcd_import(name[level:], package, level)
rstuf-api_1       |   File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
rstuf-api_1       |   File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
rstuf-api_1       |   File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
rstuf-api_1       |   File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
rstuf-api_1       |   File "<frozen importlib._bootstrap_external>", line 883, in exec_module
rstuf-api_1       |   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
rstuf-api_1       |   File "/opt/repository-service-tuf-api/app.py", line 10, in <module>
rstuf-api_1       |     from repository_service_tuf_api import settings, settings_repository
rstuf-api_1       |   File "/opt/repository-service-tuf-api/repository_service_tuf_api/__init__.py", line 98, in <module>
rstuf-api_1       |     if secrets_settings.TOKEN_KEY.startswith("/run/secrets/"):
rstuf-api_1       |   File "/usr/local/lib/python3.10/site-packages/dynaconf/base.py", line 138, in __getattr__
rstuf-api_1       |     value = getattr(self._wrapped, name)
rstuf-api_1       |   File "/usr/local/lib/python3.10/site-packages/dynaconf/base.py", line 300, in __getattribute__
rstuf-api_1       |     return super().__getattribute__(name)
rstuf-api_1       | AttributeError: 'Settings' object has no attribute 'TOKEN_KEY'
----
rstuf-api_1       | Traceback (most recent call last):
rstuf-api_1       |   File "/usr/local/bin/uvicorn", line 8, in <module>
rstuf-api_1       |     sys.exit(main())
rstuf-api_1       |   File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
rstuf-api_1       |     return self.main(*args, **kwargs)
rstuf-api_1       |   File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1055, in main
rstuf-api_1       |     rv = self.invoke(ctx)
rstuf-api_1       |   File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
rstuf-api_1       |     return ctx.invoke(self.callback, **ctx.params)
rstuf-api_1       |   File "/usr/local/lib/python3.10/site-packages/click/core.py", line 760, in invoke
rstuf-api_1       |     return __callback(*args, **kwargs)
rstuf-api_1       |   File "/usr/local/lib/python3.10/site-packages/uvicorn/main.py", line 403, in main
rstuf-api_1       |     run(
rstuf-api_1       |   File "/usr/local/lib/python3.10/site-packages/uvicorn/main.py", line 568, in run
rstuf-api_1       |     server.run()
rstuf-api_1       |   File "/usr/local/lib/python3.10/site-packages/uvicorn/server.py", line 59, in run
rstuf-api_1       |     return asyncio.run(self.serve(sockets=sockets))
rstuf-api_1       |   File "/usr/local/lib/python3.10/asyncio/runners.py", line 44, in run
rstuf-api_1       |     return loop.run_until_complete(main)
rstuf-api_1       |   File "/usr/local/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
rstuf-api_1       |     return future.result()
rstuf-api_1       |   File "/usr/local/lib/python3.10/site-packages/uvicorn/server.py", line 66, in serve
rstuf-api_1       |     config.load()
rstuf-api_1       |   File "/usr/local/lib/python3.10/site-packages/uvicorn/config.py", line 471, in load
rstuf-api_1       |     self.loaded_app = import_from_string(self.app)
rstuf-api_1       |   File "/usr/local/lib/python3.10/site-packages/uvicorn/importer.py", line 21, in import_from_string
rstuf-api_1       |     module = importlib.import_module(module_str)
rstuf-api_1       |   File "/usr/local/lib/python3.10/importlib/__init__.py", line 126, in import_module
rstuf-api_1       |     return _bootstrap._gcd_import(name[level:], package, level)
rstuf-api_1       |   File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
rstuf-api_1       |   File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
rstuf-api_1       |   File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
rstuf-api_1       |   File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
rstuf-api_1       |   File "<frozen importlib._bootstrap_external>", line 883, in exec_module
rstuf-api_1       |   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
rstuf-api_1       |   File "/opt/repository-service-tuf-api/app.py", line 10, in <module>
rstuf-api_1       |     from repository_service_tuf_api import settings, settings_repository
rstuf-api_1       |   File "/opt/repository-service-tuf-api/repository_service_tuf_api/__init__.py", line 108, in <module>
rstuf-api_1       |     if secrets_settings.ADMIN_PASSWORD.startswith("/run/secrets/"):
rstuf-api_1       |   File "/usr/local/lib/python3.10/site-packages/dynaconf/base.py", line 138, in __getattr__
rstuf-api_1       |     value = getattr(self._wrapped, name)
rstuf-api_1       |   File "/usr/local/lib/python3.10/site-packages/dynaconf/base.py", line 300, in __getattribute__
rstuf-api_1       |     return super().__getattribute__(name)
rstuf-api_1       | AttributeError: 'Settings' object has no attribute 'ADMIN_PASSWORD'
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by kairoaraujo**
This commit makes the environment variables `SECRETS_RSTUF_TOKEN_KEY` and `SECRETS_RSTUF_ADMIN_PASSWORD` a sub-settings for `RSTUF_AUTH`.

This variables are required only when the `RSTUF_AUTH` is enabled.

This PR also makes `RSTUF_AUTH` disabled by default.

Closes #324

### CODE REVIEW

**Comment by kairoaraujo** [comment]
@MVrachev 
I'm also proposing changing the default value for `RSTUF_AUTH` to False

The main reason is that it makes it very confusing to the user if the `RSTUF_AUTH` is optional but requires sub-settings.

To keep our current unit tests passing, which expects the authentication/authorization enabled, we make it default in our tests suite (tox).
If you agree with my proposal, we will need to file one new issue (low-hanging-fruit):
- [ ] Force the deployment of the service in the FT to enable the RSTUF_AUTH -- required to be merged before this PR

All the refactoring is well described in [Feature: Disable option for built-in Authentication/Authorization](https://github.com/repository-service-tuf/repository-service-tuf-api/pull/329)

File: repository_service_tuf_api/__init__.py [hunk 1]
```diff
@@ -97,60 +100,70 @@ def sync_redis():
...
-else:
-    ADMIN_PASSWORD = secrets_settings.ADMIN_PASSWORD
+if is_auth_enabled is True:
+    # Tokens
+    if secrets_settings.TOKEN_KEY.startswith("/run/secrets/"):
+        try:
+            with open(secrets_settings.TOKEN_KEY) as f:
+                SECRET_KEY = f.read().rstrip("\n")
+        except OSError as err:
+            logging.error(str(err))
```
Comment (MVrachev) [inline repository_service_tuf_api/__init__.py:114]: Maybe we can still add a small test for those lines where we have those settings set but an `OSError` is raised?

File: repository_service_tuf_api/__init__.py [hunk 2]
```diff
@@ -97,60 +100,70 @@ def sync_redis():
...
-SessionLocal = sessionmaker(autocommit=False, autoflush=False, bind=engine)
+    else:
+        SECRET_KEY = secrets_settings.TOKEN_KEY
 
+    if secrets_settings.ADMIN_PASSWORD.startswith("/run/secrets/"):
+        try:
+            with open(secrets_settings.ADMIN_PASSWORD) as f:
+                ADMIN_PASSWORD = f.read().rstrip("\n")
+        except OSError as err:
+            logging.error(str(err))
```
Comment (MVrachev) [inline repository_service_tuf_api/__init__.py:124]: Same here we can add a simple test for that case as well.

File: repository_service_tuf_api/__init__.py [hunk 3]
```diff
@@ -97,60 +100,70 @@ def sync_redis():
...
+        crud.user_add_scopes(
+            db, user, [scope for scope in crud.get_scopes(db)]
+        )
+
+    else:
+        user_scopes = [user_scope.name for user_scope in user.scopes]
+        for scope in SCOPES:
+            if scope not in user_scopes:
+                crud.user_append_scope(db, user, scope)
+                logging.info(f"Scope '{scope}' added to admin.")
```
Comment (MVrachev) [inline repository_service_tuf_api/__init__.py:166]: We haven't added a test for that before, right?
Do we want to?

**Review by MVrachev — COMMENTED** [review]
> I'm also proposing changing the default value for RSTUF_AUTH to False

I have mixed feelings about this.
I am only worried that our users can deploy RSTUF without authentication.

**Comment by kairoaraujo** [comment]
It is still a configuration and not a problem to be noticed by the user.

We don't have an “installer”. The user must deploy adding all configurations. 

It is also well-documented, and I'm working on giving more details. 

https://github.com/kairoaraujo/repository-service-tuf/tree/refactoring_guide

**Comment by MVrachev** [comment]
> It is still a configuration and not a problem to be noticed by the user.
> 
> We don't have an “installer”. The user must deploy adding all configurations.
> 
> It is also well-documented, and I'm working on giving more details.
> 
> https://github.com/kairoaraujo/repository-service-tuf/tree/refactoring_guide

I am sorry, is this should be a response to my comment about why I have mixed feelings about disabling the built-in authentication?

**Comment by kairoaraujo** [comment]
> > It is still a configuration and not a problem to be noticed by the user.
> > We don't have an “installer”. The user must deploy adding all configurations.
> > It is also well-documented, and I'm working on giving more details.
> > https://github.com/kairoaraujo/repository-service-tuf/tree/refactoring_guide
> 
> I am sorry, is this should be a response to my comment about why I have mixed feelings about disabling the built-in authentication?

Yes. I didn't quote it correctly for the answer.

File: repository_service_tuf_api/__init__.py (hunk 1 above)
Comment (MVrachev) [reply repository_service_tuf_api/__init__.py:114]: Well, it's a little different as it's about not opening a file.
My case with `AUTH` was about not being able to run the API functions without `security`.

Comment (kairoaraujo) [reply repository_service_tuf_api/__init__.py:114]: Yes, but the origins of the testing issue are the same.
The `repository_service_tuf_api` is imported by the pytest already, and I cannot overwrite/mock it to properly test.
So, I cannot `pretend.raise` the open here.

Comment (kairoaraujo) [reply repository_service_tuf_api/__init__.py:114]: Due the complication, I filed a specific issue for it
- #338

File: repository_service_tuf_api/__init__.py (hunk 2 above)
Comment (MVrachev) [reply repository_service_tuf_api/__init__.py:124]: For now, we won't fix this.
See: https://github.com/repository-service-tuf/repository-service-tuf-api/pull/329#discussion_r1182549739

File: repository_service_tuf_api/__init__.py (hunk 3 above)
Comment (MVrachev) [reply repository_service_tuf_api/__init__.py:166]: For now, we won't fix this.
See: https://github.com/repository-service-tuf/repository-service-tuf-api/pull/329#discussion_r1182549739

**Review by MVrachev — COMMENTED** [review]
> It is still a configuration and not a problem to be noticed by the user.
> 
> We don't have an “installer”. The user must deploy adding all configurations.
> 
> It is also well-documented, and I'm working on giving more details.
> 
> https://github.com/kairoaraujo/repository-service-tuf/tree/refactoring_guide

If you think it will be better to make `RSTUF_AUTH` false by default for the user, okay.
It's important that we test it by default as `true` as we do in our functional tests.
For the user, it's not such a big deal as we have documented it clearly and our authentication doesn't provide so much security.

File: docker-compose.yml [hunk 4]
```diff
@@ -92,10 +92,9 @@ services:
       - RSTUF_TOKENS_NODE=true
       - RSTUF_BROKER_SERVER="amqp://guest:guest@rabbitmq:5672"
       - RSTUF_REDIS_SERVER="redis://redis"
+      - RSTUF_AUTH=true
```
[the remark this answers is not in the document]
Comment (kairoaraujo) [reply docker-compose.yml:95]: This `docker-compose.yml` is the development environment. We want it enabled when we run the `make run-dev`.

[the remark this answers is not in the document]
Comment (MVrachev) [reply docker-compose.yml:95]: Oh, okay, you have a point.

**Review by MVrachev — APPROVED** [review]
We have decided we would merge it regardless of the coverage warnings.
We have documented the need to test them in #338.

**Pull request merged**